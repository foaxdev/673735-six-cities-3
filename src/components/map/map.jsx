import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import leaflet from "leaflet";

export class Map extends PureComponent {

  componentDidMount() {
    const icon = leaflet.icon({
      iconUrl: `img/pin.svg`,
      iconSize: [30, 40]
    });

    const map = leaflet.map(`map`, {
      center: this.props.cityCoordinates,
      zoom: 14,
      zoomControl: false,
      marker: true
    });
    map.setView(this.props.cityCoordinates, 12);

    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);

    leaflet
      .marker(this.props.offerCoordinates, {icon})
      .addTo(map);

    for (let i = 0; i < this.props.availableOffers.length; i++) {
      leaflet
        .marker([this.props.availableOffers[i][0], this.props.availableOffers[i][1]])
        .addTo(map);
    }
  }

  render() {
    return (
      <div id="map" style={{height: `100%`}}/>
    );
  }
}

Map.propTypes = {
  cityCoordinates: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
  availableOffers: PropTypes.arrayOf(
      PropTypes.arrayOf(PropTypes.number.isRequired).isRequired
  ).isRequired,
  offerCoordinates: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired
};
