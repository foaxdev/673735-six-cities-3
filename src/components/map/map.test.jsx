import React from 'react';
import renderer from 'react-test-renderer';
import {Map} from "./map";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

const TEST_CURRENT_CITY_COORDINATES = [52.38333, 4.9];
const TEST_CURRENT_CITY_MARKER_COORDINATES = [52.3709553943508, 4.89309666406198];
const TEST_COORDINATES_OF_OFFERS_BY_CITY = [
  [52.3909553943508, 4.85309666406198],
  [52.369553943508, 4.85309666406198],
  [52.3909553943508, 4.929309666406198],
  [52.3809553943508, 4.939309666406198]
];

it(`Should render main Map correctly`, () => {
  const store = mockStore({
    currentCityCoordinates: TEST_CURRENT_CITY_COORDINATES,
    coordinatesOfOffersByCity: TEST_COORDINATES_OF_OFFERS_BY_CITY,
    currentCityMarkerCoordinates: TEST_CURRENT_CITY_MARKER_COORDINATES
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Map
            currentCityCoordinates={TEST_CURRENT_CITY_COORDINATES}
            coordinatesOfOffersByCity={TEST_COORDINATES_OF_OFFERS_BY_CITY}
            currentCityMarkerCoordinates={TEST_CURRENT_CITY_MARKER_COORDINATES}/>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
