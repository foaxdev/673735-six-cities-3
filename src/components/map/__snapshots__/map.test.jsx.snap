// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render main Map correctly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;
